libs/libc/stdlib/lib_strtod.c

```diff
@@ -195,6 +196,14 @@ double strtod(FAR const char *str, FAR char **endptr)
 
       /* Process string of digits */
 
+      if (!isdigit(*p))
+        {
+          set_errno(ERANGE);
+          number = 0.0;
+          p = (FAR char *)str;
```

Comment:
why reset p to the start? should we return the error point?

Reply:
No, it must be reset.  
This is according to the standard, and it is confirmed by the Lua tests suite.

Let me explain:

`endptr` must be set to point to the first char **after** the number.  
Since parsing failed, what we are parsing is not a number. It is just a string! (That just happens to begin with something that looks like a number.)

At this point, we realize our mistake (that we try to parse the non-number / string), and the pointer is reset.  
It now correctly points to the first string character, which is also the start of our `str`, since there is no number parsed before.

Reply:
ok